recipes/spdlog/all/conanfile.py

```diff
@@ -51,98 +52,112 @@ def configure(self):
             del self.options.fPIC
 
     def requirements(self):
-        if tools.Version(self.version) >= "1.10.0":
-            self.requires("fmt/8.1.1")
-        elif tools.Version(self.version) >= "1.9.0":
-            self.requires("fmt/8.0.1")
-        elif tools.Version(self.version) >= "1.7.0":
-            self.requires("fmt/7.1.3")
-        elif tools.Version(self.version) >= "1.5.0":
-            self.requires("fmt/6.2.1")
+        # TODO: Remove in Conan 2.0 - 1.x self.requires does not support transitive_headers
+        requires = lambda ref: self.requires(ref, transitive_headers=True) if Version(conan_version) >= "2.0.0-beta" else self.requires(ref)
```

Comment:
What does `transitive_headers=True` mean?

Reply:
Indeed, I needed to ask Conan devs to understand. The fmt header is required in a public spglog header, but on the test package, fmt headers are not available by default, so to avoid the failure, we need trasitive_headers.

Reply:
We'll be working on the docs more and more.
Best reference as of know I think is this talk https://www.youtube.com/watch?v=kKGglzm5ous&t=2025

My understanding is it should be the "same" as CMake's at least in concept 🤞

Reply:
conan 1.52 will allow these arguments in `self.requires` (see https://github.com/conan-io/conan/pull/11934), but my understanding is that it won't change the behavior in conan v1 mode (everything is always public), it's just a way to be cross-compatible v1 & v2.

Reply:
That pr is directly from this work https://github.com/conan-io/conan-center-index/pull/11981#discussion_r936903281

Which is why this migration is super important 

Once 1.52 is available we cam clean this one up 👍